blueocean-dashboard/src/main/less/run-message-cell.less

```diff
@@ -1,12 +1,45 @@
 
 .RunMessageCell {
     display: flex;
+    width: 100%;
+    height: 100%;
     align-items: center;
 
-    .RunMessageCellInner {
+    .RunMessageCellInner,
+    .RunMessageCellInner > span {
+        display: flex;
         flex: 1;
-        white-space: nowrap;
-        overflow: hidden;
-        text-overflow: ellipsis;
+        width: 100%;
+        height: 100%;
+
+        a {
+            display: flex;
+            flex: 0 0 auto;
+            align-items: center;
+            height: 100%;
+            min-width: 0;
+
+            &:last-child {
+                flex-grow: 1;
+                white-space: nowrap;
+                text-overflow: ellipsis;
+                overflow: hidden;
+
+                &:not(:first-child) > span{
+                    width: 10px;
+                }
+
+                > span {
+                    display: table-cell;
+                    flex-grow: 1;
+                    vertical-align: middle;
+                    height: auto;
+                    width: 300px;
```

Comment:
Can you explain why this has width=300 on it? Is that the default width we apply to the Message column?

Reply:
in this case, the actual value of the width property does not matter as long as it is smaller than the width of the container, it can be set to "width: 0px" and it would still work the same, flex-grow: 1 makes the width as large as possible. The width needs to be set to a value for text-overflow: ellipsis to work. I have set it to 300px because of a quirk in IE11 in which in some cases the text rendered correctly but the ellipsis is rendered where the width would have finished.

Reply:
Cool. Adding a comment to that effect would be good. Whenever I need style rules for a browser-specific bug I usually add it, to discourage anyone from accidentally removing it later.